lib/logstash/filters/geoip.rb

```diff
@@ -84,6 +84,16 @@ class LogStash::Filters::GeoIP < LogStash::Filters::Base
   # number of cache misses and waste memory.
   config :lru_cache_size, :validate => :number, :default => 1000
 
+  # If the ip field is an array: check which one is not a private ip. The first non private ip will be used.
+  config :filter_private_ips, :validate => :boolean, :required => false, :default => true
+
+  # String by which the ip field should be tokenized
+  config :ip_split_symbol, :validate => :string, :required => false, :default => ","
+
+  # list of ip patterns that private ips start with. 
```

Comment:
We thought it would probably be the fastest approach, performance wise, as it allows us to do simple substring comparisons. CIDR is certainly more elegant when it comes to the notation, but I don't expect the users to modify this value, so I didn't put much weight onto that aspect. An other plus, on the other hand, would be that it makes it much easier to add the private ipv6 subnets also, so if you prefer the CIDR notation, then I'd be happy to implement that.